resources/js/Hooks/useWalletPollingIntervals.ts

```diff
@@ -0,0 +1,46 @@
+import { DateTime } from "@ardenthq/sdk-intl";
+import { LocalStorageKey, useLocalStorage } from "./useLocalStorage";
+import { isTruthy } from "@/Utils/is-truthy";
+
+export enum Interval {
+    Long = 60 * 5 * 1000, // 5 minute,
+    Short = 5 * 1000, // 5 seconds
+    Regular = 30 * 1000, // 30 seconds
+}
+
+interface CalculateIntervalProperties {
+    wallet?: App.Data.Wallet.WalletData | null;
+    lastTransactionSentAt?: number | null;
+    retries?: number;
+}
+
+interface Properties {
+    calculateInterval: (retries?: number) => number;
+}
+
+export const calculateInterval = ({ wallet, retries, lastTransactionSentAt }: CalculateIntervalProperties): number => {
+    if (
+        isTruthy(lastTransactionSentAt) &&
+        DateTime.make(lastTransactionSentAt).addMinutes(1).isBefore(DateTime.make())
+    ) {
+        return Interval.Short;
+    }
+
+    if (wallet?.timestamps.tokens_fetched_at !== null && wallet?.timestamps.native_balances_fetched_at !== null) {
+        return Interval.Long;
+    }
+
+    if (isTruthy(retries) && retries > 12) {
+        return Interval.Regular;
```

Comment:
A question for you @goga-m:
If I understood after a transaction we mark the timestamp and use it to decide at what intervals we should pull data from backend. Don't you think that we should do it after a tx minted?